src/main/php/de/thekid/dialog/processing/Watch.php

```diff
@@ -0,0 +1,111 @@
+<?php namespace de\thekid\dialog\processing;
+
+use com\mongodb\{MongoConnection, Error};
+use de\thekid\dialog\Preferences;
+use io\TempFile;
+use io\streams\StreamTransfer;
+use lang\IllegalStateException;
+use util\Objects;
+use util\cmd\{Arg, Command};
+use web\Environment;
+use webservices\rest\{Endpoint, RestUpload};
+
+/**
+ * Watch for changes on the `processing` collection and process media
+ * accordingly.
+ * 
+ * @see   https://github.com/thekid/dialog/issues/44
+ */
+class Watch extends Command {
+  private const WAIT_BEFORE_RETRYING= 10;
+  private $api;
+
+  /** Sets API endpoint to use */
+  #[Arg(position: 0)]
+  public function usingApi(?string $endpoint= null): void {
+    $this->api= new Endpoint($endpoint ?? getenv('DIALOG_API'));
+  }
+
+  /** Runs forever, consuming the change stream */
+  public function run(): int {
+    $files= new Files()
+      ->matching(['.jpg', '.jpeg', '.png', '.webp'], new Images()
+        ->targeting('preview', new ResizeTo(720, 'jpg'))
+        ->targeting('thumb', new ResizeTo(1024, 'webp'))
+        ->targeting('full', new ResizeTo(3840, 'webp'))
+      )
+      ->matching(['.mp4', '.mpeg', '.mov'], new Videos()
+        ->targeting('preview', new ResizeTo(720, 'jpg'))
+        ->targeting('thumb', new ResizeTo(1024, 'webp'))
+      )
+    ;
```

Comment:
This code is duplicated from `de.thekid.dialog.import.LocalDirectory`. Can we extract this into a trait or something similar?